### PR TITLE
refactor Footer.test.tsx using testing library

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- refactor Footer.test.tsx using testing library
  - add testing library dependencies to shared/

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Steps for QA.

## Fixes

Fixes: #3402  

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.

### CODE REVIEW

File: shared/src/components/Footer/Footer.test.tsx [hunk 1]
```diff
@@ -14,46 +14,28 @@ describe("Footer", () => {
   });
 
   it("renders", () => {
-    const wrapper = shallow(<Footer version="2.7.0" />);
-    expect(wrapper).toMatchSnapshot();
+    const { container } = render(<Footer version="2.7.0" />);
+    expect(container.firstChild).toMatchSnapshot();
```
Comment (hatched) [inline shared/src/components/Footer/Footer.test.tsx:18]: Can you remove this test? I have never found these to be very useful as we should be testing explicit features, and they always require attention when changing layout even if functionally the component still works as expected.

File: shared/src/components/Footer/Footer.test.tsx [hunk 2]
```diff
@@ -14,46 +14,28 @@ describe("Footer", () => {
...
-          (node) => node.type() === "a" && node.text() === "Give feedback"
-        )
-        .exists()
-    ).toBe(true);
+      screen.getByRole("link", { name: "Give feedback" })
+    ).toBeInTheDocument();
   });
 
   it("hides the feedback link when analytics disabled", () => {
-    const wrapper = shallow(
```
Comment (hatched) [inline shared/src/components/Footer/Footer.test.tsx:35]: Nice to get rid of those shallow renders.

**Review by hatched — APPROVED** [review]
Looks good, thanks @petermakowski can you be sure to add any lessons learned during this conversion to the `Enzyme to RTL` doc plz.

File: shared/src/components/Footer/Footer.test.tsx (hunk 1 above)
Comment (petermakowski) [reply shared/src/components/Footer/Footer.test.tsx:18]: Yeah, not a huge fan of snapshot testing either. Will remove.

Updated docs: https://docs.google.com/document/d/1ysVlPByzVY-NBUjGic1WupHojUl30P6cf3Xe-OVRRPg/edit#heading=h.48e8g53iq33

Comment (petermakowski) [reply shared/src/components/Footer/Footer.test.tsx]: Done.

**Pull request merged**